### PR TITLE
Rename fs level integration - 1

### PR DESCRIPTION
**Pull request by Tulsishah**
### Description
Create common functions to reuse it in further renameFolder method.

### Link to the issue in case of a bug fix.
NA

### Testing details
1. Manual - NA
2. Unit tests - NA
3. Integration tests - Automated

### CODE REVIEW

File: internal/fs/fs.go [hunk 1]
```diff
@@ -1972,6 +1972,45 @@ func (fs *fileSystem) renameFile(
 	return nil
 }
 
+func (fs *fileSystem) handlePendingInodes(inodes *[]inode.DirInode) func() {
```
Comment (vadlakondaswetha) [inline internal/fs/fs.go:1975]: why is this function returning a function?

File: internal/fs/fs.go [hunk 2]
```diff
@@ -1972,6 +1972,45 @@ func (fs *fileSystem) renameFile(
...
+		*inodes = []inode.DirInode{}
+	}
+}
+
+func (fs *fileSystem) getBucketDirInode(ctx context.Context, parent inode.DirInode, name string, pendingInodes *[]inode.DirInode) (inode.BucketOwnedDirInode, error) {
+	dir, err := fs.lookUpOrCreateChildDirInode(ctx, parent, name)
+	if err != nil {
+		return nil, fmt.Errorf("lookup directory: %w", err)
+	}
+	*pendingInodes = append(*pendingInodes, dir)
```
Comment (vadlakondaswetha) [inline internal/fs/fs.go:1989]: This is wrong. you are calling the method as getBucketDirInode and appending dir to pending inodes.

File: internal/fs/fs.go [hunk 3]
```diff
@@ -1989,27 +2028,15 @@ func (fs *fileSystem) renameDir(
...
 	// the kernel) and unlocks the pending inodes, but only once
 	var pendingInodes []inode.DirInode
-	releaseInodes := func() {
-		for _, in := range pendingInodes {
-			fs.unlockAndDecrementLookupCount(in, 1)
-		}
-		pendingInodes = []inode.DirInode{}
-	}
-	defer releaseInodes()
+	defer fs.handlePendingInodes(&pendingInodes)()
```
Comment (vadlakondaswetha) [inline internal/fs/fs.go:2031]: Is this working as intended. You are calling a function which is returning a function ??

File: internal/fs/fs.go [hunk 4]
```diff
@@ -1972,6 +1972,45 @@ func (fs *fileSystem) renameFile(
...
+	fs.mu.Lock()
+	entries := oldDir.LocalFileEntries(fs.localFileInodes)
+	fs.mu.Unlock()
+	if len(entries) != 0 {
+		return fmt.Errorf("can't rename directory %s with open files: %w", oldName, syscall.ENOTSUP)
+	}
+	return nil
+}
+
+func (fs *fileSystem) checkNewDirNonEmpty(newDir inode.BucketOwnedDirInode, newName string) error {
```
Comment (vadlakondaswetha) [inline internal/fs/fs.go:2003]: can you rename this to checkDirNotEmpty. Putting new in the name and the parameters is not required.

**Review by vadlakondaswetha — COMMENTED** [review]
\

File: internal/fs/fs.go (hunk 4 above)
Comment (Tulsishah) [reply internal/fs/fs.go]: Done

File: internal/fs/fs.go (hunk 2 above)
Comment (Tulsishah) [reply internal/fs/fs.go]: Done

File: internal/fs/fs.go (hunk 1 above)
Comment (Tulsishah) [reply internal/fs/fs.go]: Done

File: internal/fs/fs.go (hunk 2 above)
Comment (Tulsishah) [reply internal/fs/fs.go]: Done

File: internal/fs/fs.go (hunk 3 above)
Comment (Tulsishah) [reply internal/fs/fs.go]: Yeah... It was working correctly...but then I realized I made it unnecessarily complex. 

Fixed.

File: internal/fs/fs.go [hunk 5]
```diff
@@ -1972,6 +1972,42 @@ func (fs *fileSystem) renameFile(
...
+	if err != nil {
+		return nil, fmt.Errorf("lookup directory: %w", err)
+	}
+	return dir, nil
+}
+
+func (fs *fileSystem) checkAndHandleLocalFiles(oldDir inode.BucketOwnedDirInode, oldName string) error {
+	fs.mu.Lock()
+	entries := oldDir.LocalFileEntries(fs.localFileInodes)
+	fs.mu.Unlock()
```
Comment (vadlakondaswetha) [inline internal/fs/fs.go:1993]: nit: add a new line after unlock

File: internal/fs/fs.go [hunk 6]
```diff
@@ -1972,6 +1972,42 @@ func (fs *fileSystem) renameFile(
...
+	if len(entries) != 0 {
+		return fmt.Errorf("can't rename directory %s with open files: %w", oldName, syscall.ENOTSUP)
+	}
+	return nil
+}
+
+func (fs *fileSystem) checkDirNotEmpty(newDir inode.BucketOwnedDirInode, newName string) error {
```
Comment (vadlakondaswetha) [inline internal/fs/fs.go:2000]: Please pay attention to method names and parameter names. what does newDir and newName mean

File: internal/fs/fs.go [hunk 7]
```diff
@@ -1972,6 +1972,42 @@ func (fs *fileSystem) renameFile(
...
+	unexpected, err := newDir.ReadDescendants(context.Background(), 1)
+	if err != nil {
+		return fmt.Errorf("read descendants of the new directory %q: %w", newName, err)
```
Comment (vadlakondaswetha) [inline internal/fs/fs.go:2003]: here as well.

File: internal/fs/fs.go [hunk 8]
```diff
@@ -1972,6 +1972,42 @@ func (fs *fileSystem) renameFile(
...
+
+func (fs *fileSystem) getBucketDirInode(ctx context.Context, parent inode.DirInode, name string) (inode.BucketOwnedDirInode, error) {
+	dir, err := fs.lookUpOrCreateChildDirInode(ctx, parent, name)
+	if err != nil {
+		return nil, fmt.Errorf("lookup directory: %w", err)
+	}
+	return dir, nil
+}
+
+func (fs *fileSystem) checkAndHandleLocalFiles(oldDir inode.BucketOwnedDirInode, oldName string) error {
```
[the remark this answers is not in the document]
Comment (Tulsishah) [reply internal/fs/fs.go]: I was ensuring no local files are open file before renaming. Anyway updated name.

File: internal/fs/fs.go (hunk 6 above)
Comment (Tulsishah) [reply internal/fs/fs.go]: Yes yes... sorry...missed. 

Fixed

File: internal/fs/fs.go [hunk 9]
```diff
@@ -1972,6 +1972,44 @@ func (fs *fileSystem) renameFile(
 	return nil
 }
 
+func (fs *fileSystem) releasePendingInodes(inodes *[]inode.DirInode) {
```
Comment (vadlakondaswetha) [inline internal/fs/fs.go:1975]: rename this method to releaseInodes() -> you are passing inodes as argument. This method is not dealing with any pending inodes stuff.

**Review by vadlakondaswetha — APPROVED** [review]

**Review by vadlakondaswetha — APPROVED** [review]

**Pull request merged**